core/database/foxx/api/user_router.js

```diff
@@ -571,9 +594,71 @@ router
                         refresh: req.queryParams.refresh,
                         expiration: Math.floor(Date.now() / 1000) + req.queryParams.expires_in,
                     };
-                    g_db._update(user_id, obj, {
-                        keepNull: false,
-                    });
+
+                    switch (token_type) {
+                        // TODO: update logic for various AccessTokenType instances as they are defined.
+                        case g_lib.AccessTokenType.GENERIC: {
+                            break;
+                        }
+                        case g_lib.AccessTokenType.GLOBUS: {
+                            break;
+                        }
+                        case g_lib.AccessTokenType.GLOBUS_AUTH: {
+                            break;
+                        }
+                        case g_lib.AccessTokenType.GLOBUS_TRANSFER: {
+                            // find or insert collection
+                            const { uuid: collection_search_key, scopes } =
+                                g_lib.parseOtherTokenData(
+                                    g_lib.AccessTokenType.GLOBUS_TRANSFER,
+                                    other_token_data,
+                                ); // TODO: the call site and function and docs will need to be updated if changes are made to assumed data
+                            // GLOBUS_TRANSFER parse currently assumes uuid and scopes exist
+                            let globus_collection = g_db.globus_coll.exists({
+                                _key: collection_search_key,
+                            });
+                            if (!globus_collection) {
+                                globus_collection = g_db.globus_coll.save({
+                                    _key: collection_search_key,
+                                    name: "Newly Inserted Collection",
+                                    description: "The collection description",
+                                    required_scopes: scopes, // string - Is this always the minimum required set?
+                                    owner: "", // Should this be looked up on Globus?
+                                    ct: Math.floor(Date.now() / 1000),
+                                    ut: Math.floor(Date.now() / 1000),
+                                    type: "mapped/guest", // How do we determine this from here? Should it be passed as part of the params?
```

Comment:
You cannot determine this here, similar to scopes this will have to be passed in. As this is not a trivial amount of work I would create an issue to address it in a follow up PR. For now you can just assume it is a mapped collection.

Reply:
The only reason a Guest collection token would need to be saved is if HA is turned on and a consent flow is still required.